example/global.cc

```diff
@@ -31,164 +31,134 @@ void check(T actual, U expected) {
-  auto imports = wasm::vec<wasm::Extern*>::make(
-    const_f32_import.get(), const_i64_import.get(),
-    var_f32_import.get(), var_i64_import.get()
-  );
-  auto instance = wasm::Instance::make(store, module.get(), imports);
-  if (!instance) {
-    std::cout << "> Error instantiating module!" << std::endl;
+  wasm::Imports imports{
+    {
+      1.0f,
+      2.0,
+      var_f32_import,
+      var_i64_import,
+    },
+  };
+  auto instance = store.instantiate(file, imports);
+  if (instance.error()) {
+    std::cout << "> Error instantiating module: " << error->what() << std::endl;
     exit(1);
   }
 
   // Extract export.
   std::cout << "Extracting exports..." << std::endl;
   auto exports = instance->exports();
   size_t i = 0;
-  auto const_f32_export = get_export_global(exports, i++);
-  auto const_i64_export = get_export_global(exports, i++);
-  auto var_f32_export = get_export_global(exports, i++);
-  auto var_i64_export = get_export_global(exports, i++);
-  auto get_const_f32_import = get_export_func(exports, i++);
-  auto get_const_i64_import = get_export_func(exports, i++);
-  auto get_var_f32_import = get_export_func(exports, i++);
-  auto get_var_i64_import = get_export_func(exports, i++);
-  auto get_const_f32_export = get_export_func(exports, i++);
-  auto get_const_i64_export = get_export_func(exports, i++);
-  auto get_var_f32_export = get_export_func(exports, i++);
-  auto get_var_i64_export = get_export_func(exports, i++);
-  auto set_var_f32_import = get_export_func(exports, i++);
-  auto set_var_i64_import = get_export_func(exports, i++);
-  auto set_var_f32_export = get_export_func(exports, i++);
-  auto set_var_i64_export = get_export_func(exports, i++);
+  auto const_f32_export = instance->exported_global(i++);
+  auto const_i64_export = instance->exported_global(i++);
+  auto var_f32_export = instance->exported_global(i++);
+  auto var_i64_export = instance->exported_global(i++);
+  i = 0;
+  auto get_const_f32_import = instance->exported_func(i++);
+  auto get_const_i64_import = instance->exported_func(i++);
+  auto get_var_f32_import = instance->exported_func(i++);
+  auto get_var_i64_import = instance->exported_func(i++);
+  auto get_const_f32_export = instance->exported_func(i++);
+  auto get_const_i64_export = instance->exported_func(i++);
+  auto get_var_f32_export = instance->exported_func(i++);
+  auto get_var_i64_export = instance->exported_func(i++);
+  auto set_var_f32_import = instance->exported_func(i++);
+  auto set_var_i64_import = instance->exported_func(i++);
+  auto set_var_f32_export = instance->exported_func(i++);
+  auto set_var_i64_export = instance->exported_func(i++);
 
   // Try cloning.
-  assert(var_f32_import->copy()->same(var_f32_import.get()));
+  // TODO: what is this really doing? new global? new handle to same global?
+  // copy of data presently held by the global?
+  auto var_f32_import_copy = var_f32_import;
+  assert(var_f32_import_copy == var_f32_import);
```

Comment:
I think `same` might be a Wasm-runtime defined thing, something to keep in mind here!